electrum/wallet.py

```diff
@@ -2431,11 +2422,10 @@ def sign_recovery_transaction(self, tx: Transaction, password, recovery_keypairs
         tmp_tx = copy.deepcopy(tx)
         # update tmp tx
         self.update_transaction_multisig_generator(tmp_tx)
-
-        # this method modified tx inputs pubkeys and num_sig
         tmp_tx.add_info_from_wallet(self, include_xpubs_and_full_paths=True)
-        self.add_recovery_pubkey_to_transaction(tmp_tx)
-
+        if update_pubkeys:
```

Comment:
Could you make it more readable? Name of variable and way of use suggest it is some boolean variable but then we call it with some arguments. I was confused at first.